mender-update/daemon/state_test.cpp

```diff
@@ -2691,6 +2729,65 @@ class TestDeploymentClient : virtual public deployments::DeploymentAPI {
 	deployments::DeploymentStatus fail_status_report_status_;
 };
 
+// Normal DB, but all writes fail.
```

Comment:
This is not true for the code below.

Reply:
Indeed, it was only true when I started writing it!